util/grpc/errors.go

```diff
@@ -83,7 +86,7 @@ func kubeErrToGRPC(err error) error {
 	case apierr.IsForbidden(err):
 		err = rewrapError(err, codes.PermissionDenied)
 	case apierr.IsTimeout(err):
-		err = rewrapError(err, codes.DeadlineExceeded)
+		err = rewrapError(err, codes.DeadlineExceeded, "Command timed out before the specified conditions were met")
```

Comment:
I think this is the wrong place to make the change. The purpose of this function is to take a k8s API error and convert it to a GRPC error. In the case of an `argocd app wait` timeout, it's not the k8s API that's timing out, but rather the `argocd` client CLI.

I believe the correct way to do this will be to modify this function to capture a context timeout error and augment the error message: https://github.com/argoproj/argo-cd/blob/58138ad0a5e3951a12f407cd71fa2ebe2ae83295/cmd/argocd/commands/app.go#L2013

Reply:
Okay, yeah I got it. It should capture the timeout and throw the error with a custom message, right?